FWCore/ParameterSet/python/ModulesProxy.py

```diff
@@ -0,0 +1,24 @@
+import os
+from os.path import sep, join
+import importlib
+
+class _ModuleProxy (object):
+    def __init__(self, package, name):
+        self._package = package
+        self._name = name
+        self._caller = None
+    def __call__(self,**kwargs):
+        if not self._caller:
+            self._caller = getattr(importlib.import_module(self._package+'.'+self._name),self._name)
+        return self._caller(**kwargs)
+
+
+def _setupProxies(fullName):
+    _cmssw_package_name='.'.join(fullName.split(sep)[-3:-1])
+    filename = fullName.split(sep)[-1]
+    pathname = fullName[:-1*len(filename)]
+    proxies = dict()
+    for filename in ( x for x in os.listdir(pathname) if (x[-3:] == '.py' and x != filename and x[-6:] != 'cfi.py')):
```

Comment:
Is the intention of the second `filename` (in `x != filename`) to refer to the `filename` defined on line 18 or the loop variable?

AFAICT, it presently refers to the variable on line 18 on the first call to the generator, and to the loop variable on the subsequent calls, which feels confusing even if that would be the intention.

Reply:
The intent of `x != filename` was to always compare with the variable on line 18. However since the file name never ends with `cfi.py` the later check also works.

Reply:
fixed